### PR TITLE
make model initializer injector idempotent

### LINKED ISSUE #225: Autoscaling fails due to model initializer
/kind bug

**What steps did you take and what happened:**

Failed autoscaling example

**What did you expect to happen:**
Pods autoscaling

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version: 0.6
- KFServing Version: 0.1
- Kubeflow version:
- Minikube version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

Knative autoscaling log
```
 {"level":"info","ts":"2019-07-08T16:50:06.673Z","logger":"autoscaler.kpa-class-podautoscaler-controller","caller":"controller/controller.go:298","msg":"Reconcile failed. Time taken: 78.328181 │
│ ms.","knative.dev/controller":"kpa-class-podautoscaler-controller","knative.dev/key":"default/flowers-sample-default-6btkq"}                                                                    │
│ {"level":"info","ts":"2019-07-08T16:50:06.673Z","logger":"autoscaler.kpa-class-podautoscaler-controller.event-broadcaster","caller":"record/event.go:221","msg":"Event(v1.ObjectReference{Kind: │
│ \"PodAutoscaler\", Namespace:\"default\", Name:\"flowers-sample-default-6btkq\", UID:\"4804b980-a19b-11e9-bfb0-42010aa400e3\", APIVersion:\"autoscaling.internal.knative.dev/v1alpha1\", Resour │
│ ceVersion:\"94064\", FieldPath:\"\"}): type: 'Warning' reason: 'InternalError' error scaling target: Deployment.apps \"flowers-sample-default-6btkq-deployment\" is invalid: spec.template.spec │
│ .initContainers[1].name: Duplicate value: \"model-initializer\"","knative.dev/controller":"kpa-class-podautoscaler-controller"}
```

### PR DESCRIPTION
**Pull request by rakelkar**
Deployments were get double mutated - once on creation and again when scaled. 

Fixes #225

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/228)

### CODE REVIEW

**Comment by rakelkar** [comment]
@ellis-bigelow  @cliveseldon

File: pkg/webhook/admission/deployment/mutator.go [hunk 1]
```diff
@@ -71,6 +71,12 @@ func (mutator *Mutator) Handle(ctx context.Context, req types.Request) types.Res
 
 func (mutator *Mutator) mutate(deployment *appsv1.Deployment, configMap *v1.ConfigMap) error {
 
+	// only mutate if not already mutated
+	mutated, ok := deployment.Spec.Template.ObjectMeta.Annotations[constants.DeploymentMutatedInternalAnnotationKey]
+	if ok && mutated == "true" {
```
Comment (yuzisun) [inline pkg/webhook/admission/deployment/mutator.go]: Can we just check the number of init containers on the deployment ?

**Comment by rakelkar** [comment]
Can do.. I can move the check to the initializer and check for the presence
of the init container...

**Comment by rakelkar** [comment]
@yuzisun removed extra annotation and made the model initializer injector idempotent

File: pkg/webhook/admission/deployment/model_initializer_injector.go [hunk 2]
```diff
@@ -54,6 +54,13 @@ func (mi *ModelInitializerInjector) InjectModelInitializer(deployment *appsv1.De
 		return nil
 	}
 
+	// Dont inject if InitContianer already injected
```
Comment (ellis-bigelow) [inline pkg/webhook/admission/deployment/model_initializer_injector.go]: InitContianer type

**Comment by ellis-bigelow** [comment]
Thanks for the timely fix.

/lgtm
/approve

**Pull request merged**